airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/TestDestination.java

```diff
@@ -373,6 +392,70 @@ public void testSyncWithNormalization(String messagesFilename, String catalogFil
     assertSameMessages(messages, actualMessages, true);
   }
 
+  /**
+   * Verify that the integration successfully writes records successfully both raw and normalized and
+   * run dedup transformations.
+   */
+  @Test
+  public void testIncrementalDedupSync() throws Exception {
+    if (!implementsAppendDedup()) {
+      LOGGER.info("Destination's spec.json does not include 'append_dedup' in its '\"supportedDestinationSyncModes\"'");
+      return;
+    }
+
+    final AirbyteCatalog catalog =
+        Jsons.deserialize(MoreResources.readResource(DataArgumentsProvider.EXCHANGE_RATE_CONFIG.catalogFile), AirbyteCatalog.class);
+    final ConfiguredAirbyteCatalog configuredCatalog = CatalogHelpers.toDefaultConfiguredCatalog(catalog);
+    configuredCatalog.getStreams().forEach(s -> {
+      s.withSyncMode(SyncMode.INCREMENTAL);
+      s.withDestinationSyncMode(DestinationSyncMode.APPEND_DEDUP);
+      s.withCursorField(List.of("_airbyte_emitted_at"));
+      s.withPrimaryKey(List.of(List.of("date")));
```

Comment:
this is a good test. would it make sense to test a compound primary key as well? exchange rates is actually a good candidate for this where you can imagine the primary key being [currency, date].

Reply:
Ok, but Exchange rates connector doesn't actually include a "base" currency column (as selected in the connector's spec.json), I'll add one in the catalog/message json files then.

However moving on this idea, I added a currency value column (actual conversion rate float number) as part of the primary key. It may not make much business sense to do that but it is a failing use case in #2684